common/date.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ParseDate parses a date string of the format YYYY-MM-DD
+func ParseDate(date string) (t time.Time, err error) {
+	layout := "2006-01-02" //this represents YYYY-MM-DD
```

Comment:
If we make the layout an exposed constant, others can just call `time.Parse(cmn.TimeLayour, date)`, and then we don't need this function. I don't mind it too much though